skorch/helper.py

```diff
@@ -0,0 +1,74 @@
+"""Helper functions and classes for users.
+
+They should not be used in skorch directly.
+
+"""
+
+
+class SliceDict(dict):
+    """Wrapper for Python dict that makes it sliceable across values.
+
+    Use this if your input data is a dictionary and you have problems
+    with sklearn not being able to slice it. Wrap your dict with
+    SliceDict and it should usually work.
+
+    Example
+    -------
+    >>> X = {'key0': val0, 'key1': val1}
+    >>> search = GridSearchCV(net, params, ...)
+    >>> search.fit(X, y)  # raises error
+    >>> Xs = SliceDict(key0=val0, key1=val1)  # or Xs = SliceDict(**X)
+    >>> search.fit(Xs, y)  # works
+
+    """
+    def __init__(self, **kwargs):
+        shapes = [value.shape[0] for value in kwargs.values()]
```

Comment:
Please rename this to `lengths` instead of `shapes`. It is a bit misleading, especially in the error message.